utils/s3/client.cc

```diff
@@ -366,7 +366,7 @@ future<> client::upload_sink::finalize_upload() {
 
     s3l.trace("POST upload completion {} parts (upload id {})", _part_etags.size(), _upload_id);
     auto req = http::request::make("POST", _client->_host, _object_name);
-    req.query_parameters["uploadId"] = std::move(_upload_id);
+    req.query_parameters["uploadId"] = std::exchange(_upload_id, ""); // now upload_started() returns false
     req.write_body("xml", parts_xml_len, [this] (output_stream<char>&& out) -> future<> {
```

Comment:
we need to do the same when aborting upload, as abort_upload() can be called above when parts_xml_len == 0. unless client::upload_sink::close() is not called when the exception is triggered after abort_upload() returns.

Reply:
Yup, good catch